src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -33,10 +33,20 @@ public sealed partial class JsonObject : JsonNode
         /// <param name="options">Options to control the behavior.</param>
         public JsonObject(IEnumerable<KeyValuePair<string, JsonNode?>> properties, JsonNodeOptions? options = null) : this(options)
         {
+            bool isCaseInsensitive = IsCaseInsensitive(options);
+
+            JsonPropertyDictionary<JsonNode?> dictionary =
+                properties is ICollection<KeyValuePair<string, JsonNode?>> propertiesCollection
+                    ? new JsonPropertyDictionary<JsonNode?>(isCaseInsensitive, propertiesCollection.Count)
+                    : new JsonPropertyDictionary<JsonNode?>(isCaseInsensitive);
```

Comment:
Can be simplified a bit, perhaps collapsing a few newlines in the expression as well
```suggestion
                    ? new(isCaseInsensitive, propertiesCollection.Count)
                    : new(isCaseInsensitive);
```

Reply:
Will that syntax sugar work for the net462 target? Rider is giving me a warning saying "missing type specification" (but not an error)

Reply:
It should, it's a language feature and not a framework feature.

Reply:
https://github.com/dotnet/runtime/pull/96486/commits/3499a487af30dc942d445b6048ac9d3459936ca2